test/unit/lua_tweaks.c

```diff
@@ -241,7 +241,30 @@ main(void)
 	fiber_init(fiber_c_invoke);
 	L = luaL_newstate();
 	luaL_openlibs(L);
+
+	/* luaT_newmodule() assumes that this table exists. */
+	lua_newtable(L);
+	lua_setfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
+
 	tarantool_lua_tweaks_init(L);
+
+	/*
+	 * luaT_newmodule() assumes that tarantool has a special
+	 * loader for built-in modules. That's true, when all the
+	 * initialization code is executed. However in the unit
+	 * test we don't do that.
+	 *
+	 * Assign the module into package.loaded directly instead.
+	 *
+	 *  | local mod = loaders.builtin['internal.tweaks']
+	 *  | package.loaded['internal.tweaks'] = mod
+	 */
+	lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
+	lua_getfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
+	lua_getfield(L, -1, "internal.tweaks");
+	lua_setfield(L, -3, "internal.tweaks");
+	lua_pop(L, 2);
+
 	tarantool_lua_error_init(L);
 	fail_unless(luaT_dostring(
 			L, "tweaks = require('internal.tweaks')") == 0);
```

Comment:
I just wonder, whether this line doesn't solve the problem described above?

Reply:
Agreed to add a bit more info into the comment above.

Reply:
Added:

> In particular, tarantool_lua_init() funtcion is not called in a unit test.

Reply:
Great, thanks!